packages/database/test/exp/integration.test.ts

```diff
@@ -292,6 +337,116 @@ describe('Database@exp Tests', () => {
     expect(resolvedData.val()).to.deep.equal(initial);
   });
 
+  it('resolves get to serverCache when the database is offline', async () => {
+    const db = getDatabase(defaultApp);
+    const { writerRef } = getRWRefs(db);
+    const expected = {
+      test: 'abc'
+    };
+    await set(writerRef, expected);
+    goOffline(db);
+    const result = await get(writerRef);
+    expect(result.val()).to.deep.eq(expected);
+    goOnline(db);
+  });
+
+  it('resolves get to serverCache when the database is offline and using a parent-level listener', async () => {
+    const db = getDatabase(defaultApp);
+    const { readerRef, writerRef } = getRWRefs(db);
+    const toWrite = {
+      test: 'def'
+    };
+    const ec = EventAccumulatorFactory.waitsForExactCount(1);
+    await set(writerRef, toWrite);
+    onValue(readerRef, snapshot => {
+      ec.addEvent(snapshot);
+    });
+    await ec.promise;
+    goOffline(db);
+    const result = await get(child(readerRef, 'test'));
+    expect(result.val()).to.deep.eq(toWrite.test);
+    goOnline(db);
+  });
+
+  it('only fires listener once when calling get with limitTo', async () => {
+    const db = getDatabase(defaultApp);
+    const { readerRef, writerRef } = getRWRefs(db);
+    const ec = EventAccumulatorFactory.waitsForExactCount(1);
+    const toWrite = {
+      child1: 'test1',
+      child2: 'test2'
+    };
+    writeAndValidate(writerRef, readerRef, toWrite, ec);
+    const q = query(readerRef, limitToFirst(1));
+    const snapshot = await get(q);
+    const expected = {
+      child1: 'test1'
+    };
+    expect(snapshot.val()).to.deep.eq(expected);
+  });
+
+  it('should listen to a disjointed path and get should return the corresponding value', async () => {
+    const db = getDatabase(defaultApp);
+    const { readerRef, writerRef } = getRWRefs(db);
+    const toWrite = {
+      child1: 'test1',
+      child2: 'test2',
+      child3: 'test3'
+    };
+    const ec = EventAccumulatorFactory.waitsForExactCount(1);
+    writeAndValidate(writerRef, readerRef, toWrite, ec);
+    const otherChildrenQuery = query(
+      readerRef,
+      orderByKey(),
+      startAt('child2')
+    );
+    const expected = {
+      child2: 'test2',
+      child3: 'test3'
+    };
+    const snapshot = await get(otherChildrenQuery);
+    expect(snapshot.val()).to.deep.eq(expected);
```

Comment:
Did you mean to have another query here?
```
    const firstChildrenQuery = query(
      readerRef,
      orderByKey(),
      endAt('child1')
    );
   onValue(firstChildrenQuery, ...)
```

Reply:
Yes. Nice catch.